### ISSUE TITLE
unable to run channels aplication in examples

### ISSUE THREAD

**Issue by gurramnarayana** [meshlink/meshlink#50]
Linux 4.15.10-kali-amd64-dev #1 SMP Sat Mar 17 19:50:20 IST 2018 x86_64 GNU/Linux

```
./channels iotserver iotserver
DEBUG: meshlink_open called
DEBUG: meshlink_open returning
INFO: Joining mDNS multicast group on interface eth0.IPv6 with address fe80::a00:27ff:fe33:f3de.
INFO: New relevant interface eth0.IPv6 for mDNS.
INFO: Joining mDNS multicast group on interface eth0.IPv4 with address 10.0.2.15.
INFO: New relevant interface eth0.IPv4 for mDNS.
INFO: Network interface enumeration completed.
INFO: Registering new address record for fe80::a00:27ff:fe33:f3de on eth0.*.
INFO: Registering new address record for 10.0.2.15 on eth0.IPv4.
Chat started.
Type /help for a list of commands.
INFO: Node iotserver is part of the mesh network.
INFO: Node iotserver is part of the mesh network.
/invite iotclient
Segmentation fault
```

Please help me if iam doing any thing wrong?

**Comment by gurramnarayana** [comment]
Here is the error shown in dmesg

`channels[29530]: segfault at 0 ip 00007ff2e99a38ce sp 00007ffde2a50c28 error 4 in libc-2.27.so[7ff2e9915000+1b1000]`

**Comment by gurramnarayana** [comment]
This the messages from gdb

```
(gdb) run
Starting program: /root/meshlink/examples/channels server server
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
DEBUG: meshlink_open called
DEBUG: meshlink_open returning
[New Thread 0x7ffff7388700 (LWP 19464)]
WARNING: *** WARNING: Detected another IPv4 mDNS stack running on this host. This makes mDNS unreliable and is thus not recommended. ***
WARNING: *** WARNING: Detected another IPv6 mDNS stack running on this host. This makes mDNS unreliable and is thus not recommended. ***
INFO: Joining mDNS multicast group on interface eth0.IPv6 with address fe80::a00:27ff:fe33:f3de.
INFO: New relevant interface eth0.IPv6 for mDNS.
INFO: Joining mDNS multicast group on interface eth0.IPv4 with address 10.0.2.15.
INFO: New relevant interface eth0.IPv4 for mDNS.
INFO: Network interface enumeration completed.
INFO: Registering new address record for fe80::a00:27ff:fe33:f3de on eth0.*.
INFO: Registering new address record for 10.0.2.15 on eth0.IPv4.
[New Thread 0x7ffff6b87700 (LWP 19465)]
Chat started.
Type /help for a list of commands.
INFO: Node server is part of the mesh network.
INFO: Node server is part of the mesh network.
/invite client

Thread 1 "channels" received signal SIGSEGV, Segmentation fault.
__strcmp_sse2_unaligned () at ../sysdeps/x86_64/multiarch/strcmp-sse2-unaligned.S:32
32	../sysdeps/x86_64/multiarch/strcmp-sse2-unaligned.S: No such file or directory.
(gdb) bt
#0  __strcmp_sse2_unaligned () at ../sysdeps/x86_64/multiarch/strcmp-sse2-unaligned.S:32
#1  0x00007ffff7baa927 in get_my_hostname (mesh=0x555555758260) at meshlink.c:391
#2  meshlink_invite (mesh=0x555555758260, name=0x7fffffffdcf8 "client") at meshlink.c:1870
#3  0x0000555555555537 in parse_command (mesh=0x555555758260, buf=0x7fffffffdcf1 "invite") at channels.c:100
#4  0x0000555555555a20 in parse_input (mesh=0x555555758260, buf=0x7fffffffdcf0 "/invite") at channels.c:214
#5  0x0000555555555de0 in main (argc=3, argv=0x7fffffffe1f8) at channels.c:306
```


I think the argument of string compare is invalid somewhere or get_my_hostname is returning some inappropriate value

**Comment by gsliepen** [comment]
This was indeed a bug, that was triggered when generating invites on machines that have only IPv4 or IPv6 enabled, but not both. It's fixed in commit 87d6dff.
